src/4D_api.cpp

```diff
@@ -966,6 +966,423 @@ std::string pj_add_type_crs_if_needed(const std::string& str)
+                negative_max = max_value;
+            negative_meridian = false;
+            crossed_meridian_count++;
+        }
+        // negative meridian side max
+        if (negative_meridian
+            && (data[iii] > negative_max || negative_max == HUGE_VAL)
+            && data[iii] != HUGE_VAL
+        )
+            negative_max = data[iii];
+        // track general max value
+        if ((data[iii] > max_value || max_value == HUGE_VAL) && data[iii] != HUGE_VAL)
+            max_value = data[iii];
+    }
+    if (crossed_meridian_count == 2)
+        return negative_max;
+    else if (crossed_meridian_count == 4)
+        // bounds extends beyond -180/180
+        return 180;
+    return max_value;
+}
+
+
+// ---------------------------------------------------------------------------
+// Check if the original projected bounds contains
+// the north pole.
+// This assumes that the destination CRS is geographic.
+static bool contains_north_pole(
+    PJ* projobj,
+    PJ_DIRECTION pj_direction,
+    const double left,
+    const double bottom,
+    const double right,
+    const double top
+) {
+    double pole_y = 90;
+    double pole_x = 0;
+    proj_trans_generic(
+        projobj,
+        opposite_direction(pj_direction),
+        &pole_x, sizeof(double), 1,
+        &pole_y, sizeof(double), 1,
+        nullptr, sizeof(double), 0,
+        nullptr, sizeof(double), 0
+    );
+    if (left < pole_x && pole_x < right && top > pole_y && pole_y > bottom)
+        return true;
+    return false;
+}
+
+
+// ---------------------------------------------------------------------------
+// Check if the original projected bounds contains
+// the south pole.
+// This assumes that the destination CRS is geographic.
+static bool contains_south_pole(
+    PJ* projobj,
+    PJ_DIRECTION pj_direction,
+    const double left,
+    const double bottom,
+    const double right,
+    const double top
+) {
+    double pole_y = -90;
+    double pole_x = 0;
+    proj_trans_generic(
+        projobj,
+        opposite_direction(pj_direction),
+        &pole_x, sizeof(double), 1,
+        &pole_y, sizeof(double), 1,
+        nullptr, sizeof(double), 0,
+        nullptr, sizeof(double), 0
+    );
+    if (left < pole_x && pole_x < right && top > pole_y && pole_y > bottom)
+        return true;
+    return false;
+}
+
+
+// ---------------------------------------------------------------------------
+
+/** \brief Transform boundary,
+ *
+ * Transform boundary densifying the edges to account for nonlinear
+ * transformations along these edges and extracting the outermost bounds.
+ *
+ * If the destination CRS is geographic and right < left then the bounds
+ * crossed the antimeridian. In this scenario there are two polygons,
+ * one on each side of the antimeridian. The first polygon should be
+ * constructed with (left, bottom, 180, top) and the second with
+ * (-180, bottom, top, right).
+ *
+ * When projecting from polar projections to geographic,
```

Comment:
you probably lon, lat *input* order ? (as it is a "requirement"). But even that is annoying. For consistence, we should try to respect the axis order of the input CRS implied by the P object. The contains_north_pole/contains_south_pole should rather get the target CRS of the PJ* object with proj_get_target_crs(), check if it is geographic with proj_get_type() (if it is not, then you can give up), query the coordinate system and then the axis

Reply:
Sounds like a good idea, thanks :+1:

Reply:
That seems to do the trick (pyproj [MR](https://github.com/pyproj4/pyproj/pull/966)). Will port over to PROJ next.

Reply:
The bboxen seem to suffer from axis perplexity:  `(left, bottom, 180, top)` versus `(-180, bottom, top, right)`. I suppose `top` and `right` should be swapped in the latter case (or `180` and `top` in the former?).

Reply:
That example assumes that longitude is the first axis. It would probably be good to add an example for the scenario that latitude is the first axis.

Reply:
Oh, I see what you mean. The example needs to be fixed :-)